### PR TITLE
Add topic client

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ */
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@Slf4j
+public final class TopicClient implements Closeable {
+
+    private final @NonNull AdminClient adminClient;
+    private final @NonNull Duration timeout;
+
+    /**
+     * Create a new {@code TopicClient} using the specified configuration.
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]:
```suggestion
     * Creates a new {@code TopicClient} using the specified configuration.
```

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+     * @param topicName the topic name
+     * @param settings settings for number of partitions and replicationFactor
+     * @param config topic configuration
+     * @see #createTopic(String, TopicSettings, Map)
+     * @see #exists(String)
+     */
+    public void createIfNotExists(final String topicName, final TopicSettings settings,
+            final Map<String, String> config) {
+        if (this.exists(topicName)) {
+            log.debug("Topic {} already exists, no need to create.", topicName);
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]: Seems more like an info level log statement to me?

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                return false;
+            } else {
+                throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+            }
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+        } catch (final TimeoutException e) {
+            throw new RuntimeException("Failed to check if Kafka topic " + topicName + " exists", e);
+        }
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]: couldn't you throw the exception just once after the try-catch block? All valid responses are in early returns anyway

File: src/test/java/com/bakdata/common_kafka_streams/util/TopicClientTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    void setup() throws InterruptedException {
+        final EmbeddedKafkaConfig kafkaConfig = EmbeddedKafkaConfig.create()
+                .withNumberOfBrokers(2)
+                .build();
+        final EmbeddedKafkaClusterConfig clusterConfig = EmbeddedKafkaClusterConfig.create()
+                .provisionWith(kafkaConfig)
+                .build();
+        this.kafkaCluster = provisionWith(clusterConfig);
+        this.kafkaCluster.start();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
```
Comment (torbsto) [inline src/test/java/com/bakdata/common_kafka_streams/util/TopicClientTest.java]: It should work without the timeout

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    @Override
+    public void close() {
+        this.adminClient.close();
+    }
+
+    /**
+     * Check whether a Kafka topic exists.
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]:
```suggestion
     * Checks whether a Kafka topic exists.
```

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            final Map<String, String> config) {
+        if (this.exists(topicName)) {
+            log.debug("Topic {} already exists, no need to create.", topicName);
+        } else {
+            this.createTopic(topicName, settings, config);
+        }
+    }
+
+    /**
+     * Describe the current configuration of a Kafka topic.
```
Comment (torbsto) [inline src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]:
```suggestion
     * Describes the current configuration of a Kafka topic.
```

File: src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java (hunk 3 above)
Comment (philipp94831) [reply src/main/java/com/bakdata/common_kafka_streams/util/TopicClient.java]: We have to handle the exceptions differently. InterruptedExceptions require Thread.currentThread().interrupt();

File: src/test/java/com/bakdata/common_kafka_streams/util/TopicClientTest.java (hunk 4 above)
Comment (philipp94831) [reply src/test/java/com/bakdata/common_kafka_streams/util/TopicClientTest.java]: I'm unsure about this because I had to restart the build once...

**Pull request merged**